client/src/ui/organisms/header/index.tsx

```diff
@@ -51,8 +51,22 @@ export function Header() {
     return false;
   }
 
+  const [isProgressBarVisible, setProgressBarVisible] = useState(false);
```

Comment:
To be consistent with `[showMainMenu, setShowMainMenu]` above, you could rename these as follows:

```suggestion
  const [showProgressBar, setShowProgressBar] = useState(false);
```